File: ios/BatteryHealth.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BatteryHealth, NSObject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                withResolver:(RCTPromiseResolveBlock)resolve
                withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(getHealth:
                (RCTPromiseResolveBlock)resolve
                withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/BatteryHealth.swift
import Foundation
@objc(BatteryHealth)
class BatteryHealth: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    @objc(getHealth:withRejecter:)
    func getHealth(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        let instanceOfBatteryInfo = BatteryInfo();
        var response = Dictionary<NSString, Any>()
        response["health"] = instanceOfBatteryInfo.batteryHealth();
        response["temperature"] = instanceOfBatteryInfo.batteryTemperature();
        response["capacity"] = instanceOfBatteryInfo.batteryMaximumCapacity;
        response["voltage"] = instanceOfBatteryInfo.voltage;
        response["cycleCount"] = instanceOfBatteryInfo.batteryCycleCount;
        response["designCapacity"] = instanceOfBatteryInfo.batteryDesignCapacity;
        resolve( response )
    }
}
